logstash-core/spec/logstash/plugin_spec.rb

```diff
@@ -120,4 +120,33 @@ class LogStash::Filters::Stromae < LogStash::Filters::Base
       end
     end
   end
+
+  describe "subclass initialize" do
```

Comment:
I would expand the individual specs (codec does not have one; referee foul!) in:
`logstash/logstash-core/spec/logstash/(filter|inputs|outputs)/base_spec.rb` to do what you suggest.

These are here because it is verifying the ultimate superclass behaviour.